texparser/tar_extractor.py

```diff
@@ -8,6 +8,16 @@ def __init__(self, dataset_folder_path: str, extract_folder_path: str = "") -> N
         self.extract_folder_path = extract_folder_path
         self.dataset_folder_path = dataset_folder_path
 
+    def extract_folder(self, folder, func):
+        processes = []
+        filenames = os.listdir(folder)
+        for filename in filenames:
+            try:
+                processes.append(func(folder + "/" + filename))
+            except:
+                pass  # How should we handle already existing folders?
```

Comment:
Ignores already existing files/folders
maybe atleast print it that something is ignored